core/src/main/java/org/springframework/security/authorization/method/ReactiveExpressionUtils.java

```diff
@@ -56,7 +85,7 @@ static Mono<Boolean> evaluateAsBoolean(Expression expr, EvaluationContext ctx) {
 		});
 	}
 
-	private static Mono<Boolean> createInvalidReturnTypeMono(Expression expr) {
+	private static <T> Mono<T> createInvalidReturnTypeMono(Expression expr) {
 		return Mono.error(() -> new IllegalStateException(
 				"Expression: '" + expr.getExpressionString() + "' must return boolean or Mono<Boolean>"));
```

Comment:
I believe this message should be changed as well to include `AuthorizationDecision`